osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -43,8 +43,8 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
 
             double multiplier = PERFORMANCE_BASE_MULTIPLIER;
 
-            if (score.Mods.Any(m => m is OsuModNoFail))
-                multiplier *= Math.Max(0.90, 1.0 - 0.02 * effectiveMissCount);
+            //if (score.Mods.Any(m => m is OsuModNoFail))
```

Comment:
ignoring the other changes / issues, something like this will definitely not pass

making changes to the pp systems in general is something that is not done without heavy discussions and reasoning behind every decision (e.g. https://github.com/ppy/osu/pull/20963)